02_bubble/satish/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	// length of the array
+	n := len(s)
+
+	for swapped := true; swapped; {
+		swapped = false
+
+		for i := 1; i < n; i++ {
+			if s[i-1] > s[i] {
+				fmt.Println("Swapping in progress...")
+				// leveraging Go's tuple assignment
+				s[i], s[i-1] = s[i-1], s[i]
```

Comment:
Make a copy of `s` first, otherwise you will modify the input.
Write a test ensuring that the input does not get modified.

Reply:
Thanks! Feedback addressed.